pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1100,6 +1105,99 @@ func (r ReconcileHyperConverged) applyUpgradePatch(req *common.HcoRequest, hcoJs
 	return hcoJson, nil
 }
 
+var (
+	operatorMetrics = "hyperconverged-cluster-operator-metrics"
+	webhookMetrics  = "hyperconverged-cluster-webhook-metrics"
+
+	oldMetricsServices  map[string]corev1.Service
+	oldMetricsEndpoints map[string]corev1.Endpoints
+)
+
+func initOldMetricsServices(req *common.HcoRequest) {
+	if oldMetricsServices == nil {
+		oldMetricsServices = map[string]corev1.Service{
+			operatorMetrics: {
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      operatorMetrics,
+					Namespace: req.Namespace,
+				},
+			},
+			webhookMetrics: {
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      webhookMetrics,
+					Namespace: req.Namespace,
+				},
+			},
+		}
+	}
+}
+
+func initOldMetricsEndpoints(req *common.HcoRequest) {
+	if oldMetricsEndpoints == nil {
+		oldMetricsEndpoints = map[string]corev1.Endpoints{
+			operatorMetrics: {
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      operatorMetrics,
+					Namespace: req.Namespace,
+				},
+			},
+		}
+	}
+}
```

Comment:
I think it's can be combined into one function, then uses a single loop to delete them. WDYT?